src/transformers/models/bridgetower/image_processing_bridgetower.py

```diff
@@ -435,7 +463,34 @@ def preprocess(
                 - `"channels_first"` or `ChannelDimension.FIRST`: image in (num_channels, height, width) format.
                 - `"channels_last"` or `ChannelDimension.LAST`: image in (height, width, num_channels) format.
                 - `"none"` or `ChannelDimension.NONE`: image in (height, width) format.
+            pad_and_return_pixel_mask (`bool`, *optional*, deprecated, defaults to `self.do_pad`):
+                Whether to pad the image to the (max_height, max_width) in the batch. If `True`, a pixel mask is also
+                created and returned. Deprecated version of do_pad.
         """
+        valid_processor_keys = {
+            "images",
+            "do_resize",
+            "size",
+            "size_divisor",
+            "resample",
+            "do_rescale",
+            "rescale_factor",
+            "do_normalize",
+            "image_mean",
+            "image_std",
+            "do_pad",
+            "do_center_crop",
+            "return_tensors",
+            "data_format",
+            "input_data_format",
+            "pad_and_return_pixel_mask",
+        }
+
+        unused_keys = set(kwargs.keys()) - valid_processor_keys
+        if unused_keys:
+            unused_key_str = ", ".join(unused_keys)
+            logger.info(f"Unused or unrecognized configuration parameters: {unused_key_str}.")
```

Comment:
Good points!
- no reason at all - all for adding a similar logic to all! 
- `inspect` is a bit slow, right? But yes, also +1 to abstracting this away, I'll move it to another PR

Reply:
@amyeroberts was thinking about decorators: wdyt about having this as a wrapper/decorator in image_utils or elsewhere in utils? 
```python
        valid_processor_keys = inspect.getfullargspec(self.preprocess)[0]

        unused_keys = set(kwargs.keys()) - valid_processor_keys
        if unused_keys:
            unused_key_str = ", ".join(unused_keys)
            logger.info(f"Unused or unrecognized configuration parameters: {unused_key_str}.")
```

Reply:
LGTM! 

Perhaps instead of inspecting, we could have a class attribute with the valid_processor_keys? e.g. something like this:


```
class FooImageProcessor:
    _valid_processor_keys = ['bar', 'baz']
    ...

    def preprocess(..., kwargs):
        validate_kwargs(self._valid_processor_keys, kwargs)
```

Having a class attribute like this is something I think I'm going to end up with in the #28847 design

Reply:
sounds good yes! self-contained classes seem worth losing the decorator